src/overlays/actors/ovl_Boss_Mo/z_boss_mo.c

```diff
@@ -22,299 +111,2944 @@ const ActorInit Boss_Mo_InitVars = {
+                    } else {
+                        this->actionState = MO_TENT_READY;
+                        this->tentMaxAngle = .001f;
+                        this->tentSpeed = 0;
+                        this->swingSizeZ = 0;
+                        this->swingSizeX = 0;
+                        this->swingRateZ = 0;
+                        this->swingRateX = 0;
+                        this->timers[0] = 30;
+                    }
+                }
+                if (this->timers[0] == 4) {
+                    this->actionState = MO_TENT_READY;
+                    this->tentMaxAngle = .001f;
+                    this->tentSpeed = 0;
+                    this->swingSizeZ = 0;
+                    this->swingSizeX = 0;
+                    this->swingRateZ = 0;
+                    this->swingRateX = 0;
+                    this->timers[0] = 30;
+                }
+            }
+            if (this->actionState == MO_TENT_GRAB) {
+                player->unk_850 = 0xA;
+                player->actor.speedXZ = player->actor.velocity.y = 0;
+                Math_ApproachF(&player->actor.world.pos.x, this->grabPosRot.pos.x, 0.5f, 20.0f);
+                Math_ApproachF(&player->actor.world.pos.y, this->grabPosRot.pos.y, 0.5f, 20.0f);
+                Math_ApproachF(&player->actor.world.pos.z, this->grabPosRot.pos.z, 0.5f, 20.0f);
+                Math_ApproachS(&player->actor.shape.rot.x, this->grabPosRot.rot.x, 2, 0x7D0);
+                Math_ApproachS(&player->actor.shape.rot.y, this->grabPosRot.rot.y, 2, 0x7D0);
+                Math_ApproachS(&player->actor.shape.rot.z, this->grabPosRot.rot.z, 2, 0x7D0);
+                if (this->timers[0] == 0) {
+                    camera1 = Gameplay_GetCamera(globalCtx, MAIN_CAM);
+                    this->actionState = MO_TENT_SHAKE;
+                    this->tentMaxAngle = .001f;
+                    this->swingRateX = this->swingRateZ = this->swingSizeX = this->swingSizeZ = this->tentSpeed = 0;
+                    this->timers[0] = 150;
+                    this->mashCounter = 0;
+                    this->sfxTimer = 30;
+                    func_800F4BE8();
+                    func_80064520(globalCtx, &globalCtx->csCtx);
+                    this->csCamera = Gameplay_CreateSubCamera(globalCtx);
+                    Gameplay_ChangeCameraStatus(globalCtx, MAIN_CAM, 1);
+                    Gameplay_ChangeCameraStatus(globalCtx, this->csCamera, 7);
+                    this->cameraEye = camera1->eye;
+                    this->cameraAt = camera1->at;
+                    this->cameraYaw = Math_FAtan2F(this->cameraEye.x - this->actor.world.pos.x,
+                                                   this->cameraEye.z - this->actor.world.pos.z);
+                    this->cameraYawRate = 0;
+                    goto tent_shake; // I think this is genuinely the correct code
+                }
+            }
+            break;
+        tent_shake:
```

Comment:
```suggestion
                } else {
                    break;
                }
            } else {
                break;
            }
```
this doesn't change codegen but it changes regalloc somehow. just ignore this suggestion, putting it here if someone wanna try figure it out

Reply:
Yeah I had that before I used the goto method.